src/openforms/authentication/utils.py

```diff
@@ -53,3 +61,39 @@ def is_authenticated_with_plugin(request: Request, expected_plugin: str) -> bool
         return request.session[FORM_AUTH_SESSION_KEY]["plugin"] == expected_plugin
     except KeyError:
         return False
+
+
+def get_cosign_login_info(request: HttpRequest, form: Form) -> dict | None:
```

Comment:
I'm not a fan of bare `dict` type hints as they don't provide any information about the type of keys nor values.

In this case, since this should result in something that can be passed to the serializer I would even opt for the `LoginInfo` dataclass instead.

Note also that if you're only going to be calling this from DRF endpoints, that the correct request type hint is `rest_framework.request.Request` which wraps the Django `HttpRequest` (rather than subclassing it). If you need to support both, then it should be `HttpRequest | Request`.